crypto/engine/eng_devcrypto.c

```diff
@@ -537,11 +864,18 @@ static void prepare_digest_methods(void)
             || !EVP_MD_meth_set_cleanup(known_digest_methods[i], digest_cleanup)
             || !EVP_MD_meth_set_app_datasize(known_digest_methods[i],
                                              sizeof(struct digest_ctx))) {
+            digest_data[i].status = DEVCRYPTO_STATUS_FAILURE;
             EVP_MD_meth_free(known_digest_methods[i]);
             known_digest_methods[i] = NULL;
-        } else {
-            known_digest_nids[known_digest_nids_amount++] = digest_data[i].nid;
+            goto finish;
         }
+        digest_data[i].status = DEVCRYPTO_STATUS_USABLE;
+finish:
+        ioctl(cfd, CIOCFSESSION, &sess1.ses);
+        if (sess2.ses == 0)
```

Comment:
Are you sure?

Reply:
I remember this one.  It was a typo, I had just one =.  Then I saw it and added another one without giving it any thought.

Reply:
I'm thinking that trying to finalize session 0 might not be worthwhile, so I'm gonna guess that `==` should really be `!=`.  Correct?